tests/core/eth-module/test_transactions.py

```diff
@@ -41,6 +45,46 @@ def test_send_transaction_with_valid_chain_id(web3, make_chain_id, expect_succes
         assert 'chain ID' in str(exc_info.value)
 
 
+@pytest.mark.parametrize(
+    'to, _from, expect_success',
+    (
+        (
+            'thedao.eth',
+            'notanaddress.eth',
+            False
+        ),
+        (
+            'notanaddress.eth',
+            'thedao.eth',
+            False
+        ),
+        (
+            'rainbow.eth',
+            'thedao.eth',
+            True
+        ),
+    )
+)
+def test_send_transaction_with_ens_names(web3, to, _from, expect_success):
+    with ens_addresses(web3, [
+        ('thedao.eth', web3.eth.accounts[1]),
+        ('rainbow.eth', web3.eth.accounts[0])
+    ]):
+        transaction = {
+            'to': to,
+            'chainId': web3.eth.chainId,
+            'from': _from,
+        }
+
+        if expect_success:
```

Comment:
We've been moving away from this pattern.  It's not a big deal but separating the success and failure cases reduces the `if/else` clause context switches in the test body which can make for easier to read tests.